engine/flink/components/base/src/test/scala/pl/touk/nussknacker/engine/flink/util/transformer/ForEachTransformerSpec.scala

```diff
@@ -37,6 +42,20 @@ class ForEachTransformerSpec extends FunSuite with FlinkSpec with Matchers with
     extractResultValues(results) shouldBe List("one_1", "other_1")
   }
 
+  test("should set return type based on element types") {
+    val collectingListener = initializeListener
+    val model = modelData(List(TestRecord()), collectingListener)
+
+    val testProcess = aProcessWithForEachNode(elements = "{'one', 'other'}", resultExpression = s"#$forEachOutputVariableName + '_1'")
+
+    val compiledSources: NonEmptyList[PotentiallyStartPart] = getCompiledSources(model, testProcess)
+
+    val forEachOutputVariable = compiledSources.head.nextParts.find(_.id == forEachNodeId)
+      .get.validationContext.variables(forEachOutputVariableName)
+
+    forEachOutputVariable.asInstanceOf[typing.TypedClass].klass shouldBe classOf[String]
```

Comment:
ca be just `shouldBe Typed[String]` without extracting klass?

Reply:
sure, wasn't aware it's possible like that :D